File: sponge-examples/examples/script/js/events_removing.js
/**
 * Sponge Knowledge base
 * Removing scheduled events
 */

var AtomicInteger = java.util.concurrent.atomic.AtomicInteger;

function onInit() {
    EPS.setVariable("eventEntry", null);
    EPS.setVariable("eventCounter", new AtomicInteger(0));
    EPS.setVariable("allowNumber", 3);
}

var Trigger1 = Java.extend(Trigger, {
    configure: function(self) {
        self.eventName = "e1";
    },
    run: function(self, event) {
        eventCounter = EPS.getVariable("eventCounter")
    	eventCounter++;
        EPS.setVariable("eventCounter", eventCounter);
        self.logger.debug("Received event {}, counter: {}", event.name, eventCounter);
        if (eventCounter > EPS.getVariable("allowNumber")) {
        	self.logger.debug("This line should not be displayed!");
        }
    }
});

var Trigger2 = Java.extend(Trigger, {
    configure: function(self) {
        self.eventName = "e2";
    },
    run: function(self, event) {
        self.logger.debug("Removing entry");
        EPS.removeEvent(EPS.getVariable("eventEntry"));
    }
});

function onStartup() {
    var start = 100;
    var interval = 500;
    EPS.setVariable("eventEntry", EPS.event("e1").sendAfter(start, interval));
    EPS.event("e2").sendAfter(interval * EPS.getVariable("allowNumber"));
}

File: sponge-examples/examples/script/rb/events_removing.rb
# Sponge Knowledge base
# Removing scheduled events

java_import java.util.concurrent.atomic.AtomicInteger

def onInit
    $eventEntry = nil
    $eventCounter = AtomicInteger.new(0)
    $EPS.setVariable("eventCounter", $eventCounter)
    $EPS.setVariable("allowNumber", 3)
end

class Trigger1 < Trigger
    def configure
        self.eventName = "e1"
    end
    def run(event)
    	$eventCounter.incrementAndGet
        self.logger.debug("Received event {}, counter: {}", event.name, $eventCounter)
        if $eventCounter.get() > $EPS.getVariable("allowNumber")
        	self.logger.debug("This line should not be displayed!")
        end
    end
end

class Trigger2 < Trigger
    def configure
        self.eventName = "e2"
    end
    def run(event)
        self.logger.debug("Removing entry")
        $EPS.removeEvent($eventEntry)
    end
end

def onStartup
    start = 100
    interval = 500
    $eventEntry = $EPS.event("e1").sendAfter(start, interval)
    $EPS.event("e2").sendAfter(interval * $EPS.getVariable("allowNumber"))
end

